### PR TITLE
Added stream iteration unit test - issue 99

### PR DESCRIPTION
**Pull request by Drioastorga**
I have added a unit test for the stream iteration.

### CODE REVIEW

File: include/native/stream_iteration.h [hunk 1]
```diff
@@ -157,17 +157,51 @@ template<typename GenFunc, typename Operation, typename Predicate, typename OutF
...
+     do{
+       val = f(val);
+     }while(condition(val));
+     produced_queue.push(std::experimental::optional<produced_type>{val});
+     item = generated_queue.pop();
+    }
+    produced_queue.push(std::experimental::optional<produced_type>{});
+  });
+
+  std::thread sink([&cons,&produced_queue](){
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: Rename sink->consumer_task

File: include/native/stream_iteration.h [hunk 2]
```diff
@@ -157,17 +157,51 @@ template<typename GenFunc, typename Operation, typename Predicate, typename OutF
 
 }
 
+template<typename Generator, typename Operation, typename Predicate, typename Consumer>
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: Rename Operation->Transformer

File: include/native/stream_iteration.h [hunk 3]
```diff
@@ -157,17 +157,51 @@ template<typename GenFunc, typename Operation, typename Predicate, typename OutF
...
+  using generated_value_type = typename generated_type::value_type;
+
+  using produced_type = typename result_of<Operation(generated_value_type)>::type;
+  
+  mpmc_queue<generated_type> generated_queue{ex.queue_size,ex.lockfree};
+  mpmc_queue<std::experimental::optional<produced_type>> produced_queue{ex.queue_size,ex.lockfree};
+
+
+  std::thread prod([&gen, &generated_queue](){
+    while(1){
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: for (;;)

File: include/native/stream_iteration.h [hunk 4]
```diff
@@ -157,17 +157,51 @@ template<typename GenFunc, typename Operation, typename Predicate, typename OutF
...
+      auto k = gen();
+      generated_queue.push(k);
+      if(!k) break;
+    }
+  });
+  
+  std::thread task([&generated_queue,&f,&condition,&produced_queue](){
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: Rename task->transformer_task

File: include/native/stream_iteration.h [hunk 5]
```diff
@@ -157,17 +157,51 @@ template<typename GenFunc, typename Operation, typename Predicate, typename OutF
...
+  using generated_type = typename result_of<Generator()>::type;
+  using generated_value_type = typename generated_type::value_type;
+
+  using produced_type = typename result_of<Operation(generated_value_type)>::type;
+  
+  mpmc_queue<generated_type> generated_queue{ex.queue_size,ex.lockfree};
+  mpmc_queue<std::experimental::optional<produced_type>> produced_queue{ex.queue_size,ex.lockfree};
+
+
+  std::thread prod([&gen, &generated_queue](){
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: Rename prod -> producer_task

File: include/native/stream_iteration.h [hunk 6]
```diff
@@ -157,17 +157,51 @@ template<typename GenFunc, typename Operation, typename Predicate, typename OutF
 
 }
 
+template<typename Generator, typename Operation, typename Predicate, typename Consumer>
+void stream_iteration(parallel_execution_native &ex, Generator && gen, Operation && f, Predicate && condition, Consumer && cons){
+  using namespace std;
+  using generated_type = typename result_of<Generator()>::type;
+  using generated_value_type = typename generated_type::value_type;
+
+  using produced_type = typename result_of<Operation(generated_value_type)>::type;
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: transformed_type

File: include/native/stream_iteration.h [hunk 7]
```diff
@@ -157,17 +157,51 @@ template<typename GenFunc, typename Operation, typename Predicate, typename OutF
...
+  
+  mpmc_queue<generated_type> generated_queue{ex.queue_size,ex.lockfree};
+  mpmc_queue<std::experimental::optional<produced_type>> produced_queue{ex.queue_size,ex.lockfree};
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: At the begining of function put 

~~~c++
using namespace std::experimental
~~~

Then you simply use `optional<T>`

File: include/native/stream_iteration.h [hunk 8]
```diff
@@ -157,17 +157,51 @@ template<typename GenFunc, typename Operation, typename Predicate, typename OutF
...
+
+
+  std::thread prod([&gen, &generated_queue](){
+    while(1){
+      auto k = gen();
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: k -> item

File: include/native/stream_iteration.h [hunk 9]
```diff
@@ -157,17 +157,51 @@ template<typename GenFunc, typename Operation, typename Predicate, typename OutF
...
+      generated_queue.push(k);
+      if(!k) break;
+    }
+  });
+  
+  std::thread task([&generated_queue,&f,&condition,&produced_queue](){
+    auto item = generated_queue.pop();
+    while(item){
+     produced_type val = *item;
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]:
~~~c++
auto val = *item;
~~~

File: include/native/stream_iteration.h [hunk 10]
```diff
@@ -157,17 +157,51 @@ template<typename GenFunc, typename Operation, typename Predicate, typename OutF
...
+     do{
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: blank between do and open-brace

File: include/native/stream_iteration.h [hunk 11]
```diff
@@ -157,17 +157,51 @@ template<typename GenFunc, typename Operation, typename Predicate, typename OutF
...
+       val = f(val);
+     }while(condition(val));
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: Blank between close brace and `while`

File: include/native/stream_iteration.h [hunk 12]
```diff
@@ -157,17 +157,51 @@ template<typename GenFunc, typename Operation, typename Predicate, typename OutF
...
+     produced_queue.push(std::experimental::optional<produced_type>{val});
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: No type qualifier needed here.

File: include/native/stream_iteration.h [hunk 13]
```diff
@@ -157,17 +157,51 @@ template<typename GenFunc, typename Operation, typename Predicate, typename OutF
 
 }
 
+template<typename Generator, typename Operation, typename Predicate, typename Consumer>
+void stream_iteration(parallel_execution_native &ex, Generator && gen, Operation && f, Predicate && condition, Consumer && cons){
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: argumentes names:
* `generate_op`
* `transform_op`
* `predicate_op`
* `consume_op`

File: include/seq/stream_iteration.h [hunk 14]
```diff
@@ -29,9 +29,9 @@ template<typename GenFunc, typename Operation, typename Predicate, typename OutF
    while(1){
        auto k = in();
        if(!k) break;
-       auto val = k.value();
+       auto val = *k;
```
Comment (jdgarciauc3m) [inline include/seq/stream_iteration.h]: k -> item

**Review by jdgarciauc3m — CHANGES_REQUESTED** [review]
See comments in code.

File: include/native/stream_iteration.h [hunk 15]
```diff
@@ -29,145 +29,179 @@
...
-      p.register_thread();
-      while(1){
-         auto k = in();
-         if(!k){
-             sendFinish=true;
-             break;
-         }
-         nelem++;
-         queue.push(k);
+template<typename Generator, typename Predicate, typename Consumer, typename ...Stages>
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: Stages -> MoreTransformers

File: include/native/stream_iteration.h [hunk 16]
```diff
@@ -29,145 +29,179 @@
...
+void stream_iteration(parallel_execution_native &ex, Generator && generate_op, pipeline_info<parallel_execution_native , Stages...> && pipe, Predicate && predicate_op, Consumer && consume_op){
+  mpmc_queue< typename std::result_of<Generator()>::type > queue(ex.queue_size,ex.lockfree);
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: Factor-out  `generated_type`

Comment (jdgarciauc3m) [reply include/native/stream_iteration.h]: Use uniform initializer syntax.

File: include/native/stream_iteration.h [hunk 17]
```diff
@@ -29,145 +29,179 @@
...
+  mpmc_queue< typename std::result_of<Generator()>::type > queueOut(ex.queue_size,ex.lockfree);
+  std::atomic<int> nend (0);
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: done_threads

Comment (jdgarciauc3m) [reply include/native/stream_iteration.h]: Uniform syntax.

File: include/native/stream_iteration.h [hunk 18]
```diff
@@ -29,145 +29,179 @@
...
+  std::atomic<int> nelem (0);
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: Uniform syntax

File: include/native/stream_iteration.h [hunk 19]
```diff
@@ -29,145 +29,179 @@
...
+  std::atomic<bool> send_finish( false );
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: Uniform syntax.

File: include/native/stream_iteration.h [hunk 20]
```diff
@@ -29,145 +29,179 @@
...
+      for (;;){
+         auto item = queue_out.pop();
+         if(!item) break;
+         consume_op(*item);
       }
    });
    //Join threads
    auto first = tasks.begin();
    auto end = tasks.end();
    for(;first!=end;first++) (*first).join();
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: Can be a range-based for loop.

Comment (jdgarciauc3m) [reply include/native/stream_iteration.h]:
~~~c++
first->join()
~~~

File: include/native/stream_iteration.h [hunk 21]
```diff
@@ -29,145 +29,179 @@
...
-   std::vector<std::thread> pipeThreads;
-   composed_pipeline< mpmc_queue< typename std::result_of<GenFunc()>::type >, mpmc_queue< typename std::result_of<GenFunc()>::type >, 0, Stages ...>
-      (queue, std::forward<pipeline_info<parallel_execution_native , Stages...> >(se) , queueOut, pipeThreads); 
+      nelem++;
+      queue.push(item);
+    }
+    ex.deregister_thread();
+  });
+
+  std::vector<std::thread> pipe_threads;
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: With 

~~~c++
using namespace std
~~~

explicit qualification is not needed

File: include/native/stream_iteration.h [hunk 22]
```diff
@@ -29,145 +29,179 @@
...
+  composed_pipeline< mpmc_queue< typename std::result_of<Generator()>::type >, mpmc_queue< typename std::result_of<Generator()>::type >, 0, Stages ...>
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: Factor out typedefs with using

File: include/native/stream_iteration.h [hunk 23]
```diff
@@ -29,145 +29,179 @@
...
-
-   }
-   auto first = pipeThreads.begin();
-   auto end = pipeThreads.end();
-   gen.join();
-   for(;first!=end;first++){ (*first).join();}
-
+  for (;;) {
+    //If every element has been processed
+    if(send_finish&&nelem==0){
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]:
~~~c++
if (send_finish && nelem==0) {
~~~

File: include/native/stream_iteration.h [hunk 24]
```diff
@@ -29,145 +29,179 @@
...
+      queue.push({});
+      send_finish= false;
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]:
~~~c+++
send_finish = false;
~~~

File: include/native/stream_iteration.h [hunk 25]
```diff
@@ -29,145 +29,179 @@
...
+      break;
+    }
+    auto item = queueOut.pop();
+    //Check the predicate
+    if( !predicate_op(*item ) ) {
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]:
~~~c++
if (!predicate_op(*item )) {
~~~

File: include/native/stream_iteration.h [hunk 26]
```diff
@@ -29,145 +29,179 @@
...
+      nelem--;
+      consume_op( *item );
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]:
~~~c++
consume_op(*item);
~~~

File: include/native/stream_iteration.h [hunk 27]
```diff
@@ -29,145 +29,179 @@
...
+      //If the condition is not met reintroduce the element in the input queue
+    }else queue.push( item );
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]:
~~~c++
} else queue.push(item);
~~~

File: include/native/stream_iteration.h [hunk 28]
```diff
@@ -29,145 +29,179 @@
...
-   for(;first!=end;first++){ (*first).join();}
-
+  for (;;) {
+    //If every element has been processed
+    if(send_finish&&nelem==0){
+      queue.push({});
+      send_finish= false;
+      break;
+    }
+    auto item = queueOut.pop();
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]:
~~~c++
auto item{queueOut.pop()};
~~~

File: include/native/stream_iteration.h (hunk 16 above)
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: queue -> generated_queue

File: include/native/stream_iteration.h [hunk 29]
```diff
@@ -29,145 +29,179 @@
...
 }
 
-template<typename GenFunc, typename Operation, typename Predicate, typename OutFunc>
- void stream_iteration(parallel_execution_native &p, GenFunc && in, farm_info<parallel_execution_native,Operation> && se, Predicate && condition, OutFunc && out){
+template<typename Generator, typename Transformer, typename Predicate, typename Consumer>
+void stream_iteration(parallel_execution_native &ex, Generator && generate_op, farm_info<parallel_execution_native,Transformer> && farm, Predicate && predicate_op, Consumer && consume_op){
    std::vector<std::thread> tasks;
-   mpmc_queue< typename std::result_of<GenFunc()>::type > queue(p.queue_size,p.lockfree);
-   mpmc_queue< typename std::result_of<GenFunc()>::type > queueOut(p.queue_size,p.lockfree);
+   mpmc_queue< typename std::result_of<Generator()>::type > queue(ex.queue_size,ex.lockfree);
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: generated_queue

**Review by jdgarciauc3m — CHANGES_REQUESTED** [review]
See comments

File: include/native/stream_iteration.h [hunk 30]
```diff
@@ -29,145 +29,179 @@
...
-             sendFinish=true;
-             break;
-         }
-         nelem++;
-         queue.push(k);
+template<typename Generator, typename Predicate, typename Consumer, typename ...MoreTransformers>
+void stream_iteration(parallel_execution_native &ex, Generator && generate_op, pipeline_info<parallel_execution_native , MoreTransformers...> && pipe, Predicate && predicate_op, Consumer && consume_op){
+  using namespace std;
+  using generated_type = typename std::result_of<Generator()>::type;
+  mpmc_queue< generated_type > generated_queue{ex.queue_size,ex.lockfree};
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]:
~~~c++
mpmc_queue<generated_type>
~~~

File: include/native/stream_iteration.h [hunk 31]
```diff
@@ -29,145 +29,179 @@
...
+  mpmc_queue< generated_type > transformed_queue{ex.queue_size,ex.lockfree};
+  std::atomic<int> nelem {0};
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: nelem -> num_elements

File: include/native/stream_iteration.h [hunk 32]
```diff
@@ -29,145 +29,179 @@
...
+  std::atomic<bool> send_finish {false};
+  //Stream generator
+  thread generator_task([&](){
+    // Register the thread in the execution model
+    ex.register_thread();
+    for (;;) {
+      auto item = generate_op();
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]:
~~~c++
auto item{generate_op()};
~~~

File: include/native/stream_iteration.h [hunk 33]
```diff
@@ -29,145 +29,179 @@
...
-      (queue, std::forward<pipeline_info<parallel_execution_native , Stages...> >(se) , queueOut, pipeThreads); 
+      nelem++;
+      generated_queue.push(item);
+    }
+    ex.deregister_thread();
+  });
+
+  vector<thread> pipe_threads;
+  composed_pipeline< mpmc_queue<generated_type>, mpmc_queue<generated_type>, 0, MoreTransformers ...>
+    (generated_queue, forward<pipeline_info<parallel_execution_native , MoreTransformers...> >(pipe) , transformed_queue, pipe_threads);
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]:
~~~c++
using pipeline_info_type = pipeline_info<parallel_execution_native , MoreTransformers...>;
~~~

File: include/native/stream_iteration.h [hunk 34]
```diff
@@ -29,145 +29,179 @@
...
+      send_finish = false;
+      break;
+    }
+    auto item{transformed_queue.pop()};
+    //Check the predicate
+    if (!predicate_op(*item)) {
+      nelem--;
+      consume_op(*item);
+      //If the condition is not met reintroduce the element in the input queue
+    }else queue.push(item);
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]:
~~~c++
} 
else {
  queue.push(item);
}

File: include/native/stream_iteration.h [hunk 35]
```diff
@@ -29,145 +29,179 @@
...
- void stream_iteration(parallel_execution_native &p, GenFunc && in, farm_info<parallel_execution_native,Operation> && se, Predicate && condition, OutFunc && out){
-   std::vector<std::thread> tasks;
-   mpmc_queue< typename std::result_of<GenFunc()>::type > queue(p.queue_size,p.lockfree);
-   mpmc_queue< typename std::result_of<GenFunc()>::type > queueOut(p.queue_size,p.lockfree);
-   std::atomic<int> nend (0);
+template<typename Generator, typename Transformer, typename Predicate, typename Consumer>
+void stream_iteration(parallel_execution_native &ex, Generator && generate_op, farm_info<parallel_execution_native,Transformer> && farm, Predicate && predicate_op, Consumer && consume_op){
+  using namespace std;
+  using generated_type = typename std::result_of<Generator()>::type;
+  mpmc_queue< generated_type > generated_queue{ex.queue_size,ex.lockfree};
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]:
~~~c++
mpmc_queue<generated_type> generated_queue{ex.queue_size,ex.lock_free};
~~~

File: include/native/stream_iteration.h [hunk 36]
```diff
@@ -29,145 +29,179 @@
...
+  mpmc_queue< generated_type > transformed_queue{ex.queue_size,ex.lockfree};
+  atomic<int> done_threads {0};
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]:
~~~c++
atomic<int> done_threads{0};
~~~

File: include/native/stream_iteration.h [hunk 37]
```diff
@@ -29,145 +29,179 @@
...
+  //Output function
+  std::thread consumer_task([&](){
+    for (;;){
+     auto item = queue_out.pop();
+     if(!item) break;
+       consume_op(*item);
+     }
+  });
+  //Join threads
+  for(auto && t : tasks) t.join();
```
Comment (jdgarciauc3m) [inline include/native/stream_iteration.h]: for(auto && t : tasks) { t.join(); }

File: unit_tests/stream_iteration.cpp [hunk 38]
```diff
@@ -0,0 +1,293 @@
...
+      return val<10;
+    },
+    [this](int val) {
+      this->invocations_cons++;
+      this->out += val;
+    }
+  );
+  this->check_composed_farm();
+}
+/*
```
Comment (jdgarciauc3m) [inline unit_tests/stream_iteration.cpp]: Why is this comented out?

File: unit_tests/stream_iteration.cpp [hunk 39]
```diff
@@ -0,0 +1,293 @@
...
+      return val<10;
+    },
+    [this](int val) {
+      this->invocations_cons++;
+      this->out += val;
+    }
+  );
+  this->check_composed_pipeline();
+}
+/*
```
Comment (jdgarciauc3m) [inline unit_tests/stream_iteration.cpp]: Why is this comented out?

File: unit_tests/stream_iteration.cpp [hunk 40]
```diff
@@ -0,0 +1,293 @@
...
+      return val<10;
+    },
+    [this](int val) {
+      this->invocations_cons++; 
+      this->out += val;
+    }
+  );
+  this->check_no_composed();
+}
+/*
```
Comment (jdgarciauc3m) [inline unit_tests/stream_iteration.cpp]: Why is this comented out?

File: include/seq/stream_iteration.h [hunk 41]
```diff
@@ -24,57 +24,57 @@
...
-       auto k = in();
-       if(!k) break; 
-       auto val = k.value();
-       do{
-             val = composed_pipeline<typename std::result_of<GenFunc()>::type::value_type,0,Stages...>(val,std::forward<pipeline_info<sequential_execution, Stages...>>(f) );
-       }while(condition(val));
-       out(val);
-   }
-  
+template<typename Generator, typename Predicate, typename Consumer, typename ...Stages>
```
Comment (jdgarciauc3m) [inline include/seq/stream_iteration.h]: doxygen?

File: include/seq/stream_iteration.h [hunk 42]
```diff
@@ -24,57 +24,57 @@
...
+      val = farm.task(val);
+    } while (predicate_op(val));
+    consume_op(val);
+  }
 }
 
-template<typename GenFunc, typename Predicate, typename OutFunc, typename ...Stages>
- void stream_iteration(sequential_execution &s, GenFunc && in, pipeline_info<sequential_execution, Stages...> & f, Predicate && condition, OutFunc && out){
-    stream_iteration(s, std::forward<GenFunc>(in), std::forward<pipeline_info<sequential_execution, Stages...> &&>( f ), std::forward<Predicate>(condition), std::forward< OutFunc>( out ));
+template<typename Generator, typename Predicate, typename Consumer, typename ...Stages>
```
Comment (jdgarciauc3m) [inline include/seq/stream_iteration.h:96]: doxygen?

File: include/seq/stream_iteration.h [hunk 43]
```diff
@@ -24,57 +24,57 @@
...
-   while(1){
-       auto k = in();       
-       if(!k) break;
-       auto val = k.value();
-       do{
-          val = f.task(val);
-       } while(condition(val));
-       out(val);
-   }
+template<typename Generator, typename Transformer, typename Predicate, typename Consumer>
```
Comment (jdgarciauc3m) [inline include/seq/stream_iteration.h]: doxygen?

File: include/seq/stream_iteration.h [hunk 44]
```diff
@@ -24,57 +24,57 @@
...
+    consume_op(val);
+  }
 }
 
 
-template<typename GenFunc, typename Operation, typename Predicate, typename OutFunc>
- void stream_iteration(sequential_execution &s, GenFunc && in, farm_info<sequential_execution, Operation> & f, Predicate && condition, OutFunc && out){
-    stream_iteration(s, std::forward<GenFunc>( in ), std::forward<farm_info<sequential_execution, Operation> &&>( f ), 
-          std::forward<Predicate>( condition), std::forward< OutFunc >( out ) );
+template<typename Generator, typename Transformer, typename Predicate, typename Consumer>
```
Comment (jdgarciauc3m) [inline include/seq/stream_iteration.h:64]: doxygen?

File: include/seq/stream_iteration.h [hunk 45]
```diff
@@ -24,57 +24,57 @@
...
-   while(1){
-       auto k = in();
-       if(!k) break;
-       auto val = k.value();
-       do{
-          val = f(val);
-       }while(condition(val));
-       out(val);
-   }
+template<typename Generator, typename Transformer, typename Predicate, typename Consumer>
```
Comment (jdgarciauc3m) [inline include/seq/stream_iteration.h]: doxygen?

File: unit_tests/stream_iteration.cpp (hunk 38 above)
Comment (Drioastorga) [reply unit_tests/stream_iteration.cpp]: This is commented because de polymorphic has not the stream iteration implemented yet.

**Review by jdgarciauc3m — CHANGES_REQUESTED** [review]
More comments to address...

**Review by jdgarciauc3m — CHANGES_REQUESTED** [review]
poly/stream_iteration.h needs to be revised to select only between sequential and native implementations.

Tests should be revised to include testing of poly stream_iteration.

**Comment by Drioastorga** [comment]
I have solved the issue and now the stream iteration polymorphic execution version can be composed with farm and pipeline ,and the unit test covers it.

**Comment by jdgarciauc3m** [comment]
The following tests FAILED:
	 33 - stream_iteration_test/*.static_composed_pipeline (OTHER_FAULT)
	 34 - stream_iteration_test/*.poly_composed_pipeline (OTHER_FAULT)
	 75 - composed_reduce_test/*.static_window_offset (Failed)
	 76 - composed_reduce_test/*.static_offset_window (Failed)

**Comment by Drioastorga** [comment]
Fixed.

**Review by jdgarciauc3m — APPROVED** [review]
OK

**Comment by jdgarciauc3m** [comment]
Closes issue #99

**Pull request merged**